libr/arch/p/sbpf_cs/plugin.c

```diff
@@ -95,22 +185,394 @@ static bool decode(RArchSession *a, RAnalOp *op, RArchDecodeMask mask) {
 	const ut8 *buf = op->bytes;
 	const int len = op->size;
 	op->size = 8;
+
+	// Detect sBPF version once at the start
+	ut32 sbpf_version = detect_sbpf_version (a);
+
 	cs_insn *insn = NULL;
 	int n = cs_disasm (cpd->cs_handle, (ut8*)buf, len, op->addr, 1, &insn);
 	if (n < 1) {
-		op->type = R_ANAL_OP_TYPE_ILL;
-		if (mask & R_ARCH_OP_MASK_DISASM) {
-			op->mnemonic = strdup ("invalid");
+		// Check for instructions that Capstone doesn't decode correctly
+		// In sBPF V3 opcodes for exit and syscall seems that are swapped
+		if (len >= 8 && buf[0] == SBPF_INS_EXIT_V3 && sbpf_version >= SBPF_V3) {
```

Comment:
the sbpf_version is the hard conditional here. move all this conditionals into a separate function so the code in here is easy and short to read. and all the sbpf versioned checks happens outside